docs/enhancements/api-for-skipping-validations.md

```diff
@@ -0,0 +1,139 @@
+---
+title: api-for-skipping-validations
+authors:
+  - "@pmaidmen"
+creation-date: 2023-01-08
+last-updated: 2023-01-08
+---
+
+# Allow validations to be skipped
+
+# Summary
+
+We need new api that will allow us to skip cluster/host validations.
+This api should have it's own feature flag to allow us to disable it.
```

Comment:
*its
in addition to a feature-flag, maybe we'll want to have the same ability as multi-arch images of specifying what domains can do this skipping action?
an advantage is having this thing scoped only for whatever company needs it. the disadvantage - showing an API that is blocked for certain users is problematic. wdyt?

Reply:
One of the motivations for this feature is (as worded in the original ticket) "Sometimes we have bugs in our validations that block people from installing and we don't want our partners to stuck cause of us"

It strikes me that we are likely to advise a partner to skip a validation when we are aware of an issue and we are also aware that a workaround cannot be immediately implemented. I would imagine that this would require more flexibility. Locking this feature to specific users might cause issues with this.

Though, adding the _ability_ to lock this feature to a specific set of beta testers might not be a bad idea?

Reply:
I'm divided on this, because as you say this mechanism removes some of the flexibility here (which is key in this whole feature)
I think all-in-all it's not worth locking to users. It requires more work to be implemented, complicates the system, and removes the flexibility. I'm withdrawing my idea :sweat_smile:

Reply:
Actually maybe we need to open it for certain partners and us, at least for now, like nvidia, marvel ?
We should allow it though in all onprem solutions

Reply:
It's possible. But keep in mind that approving a new partner in SaaS prod requires a MR approval from an outside team (meaning we don't have full control over it)

Reply:
if we have set for specific orgs then it will be enabled automatically for on prem no?

Reply:
OK, we will add "per customer feature enablement" as per https://github.com/openshift/assisted-service/blob/master/docs/dev/feature-per-customer.md